engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -3129,6 +3129,9 @@ static int CreateBuffer(lua_State* L)
         buffer = dst_buffer;
     }
 
+    // before overwriting the resource buffer we just created, we need to garbage collect it
+    dmBuffer::Destroy(resource->m_Buffer);
```

Comment:
On previous lines(https://github.com/defold/defold/pull/10413/files#diff-b769dc0dca4b39eee99745dc078794a1924b15022b0ada4728fbab617c91d723R3086-R3105) we create a resource buffer using a temp data structure.
However, we don't garbage collect that created buffer before we overwrite it.